cores/arduino/WString.cpp

```diff
@@ -492,7 +492,13 @@ int String::compareTo(const String &s) const
 		if (buffer && len > 0) return *(unsigned char *)buffer;
 		return 0;
 	}
-	return strcmp(buffer, s.buffer);
+	const char *p1 = buffer;
+	const char *p2 = s.buffer;
+
+	while (*p1 == *p2++)
+		if ('\0' == *p1++)
+			return 0;
+	return (*(const char *)p1 - *(const char *)(p2 - 1));
```

Comment:
Sneaking in a functional change ;)
I'm happy with this, usually not a fan of re-implementing std routines but in this case it does appear that
"strcmp" is indeed not behaving as advertised. I'll fix the conflict and merge it.

This seems odd to me since we are not using any home-made versions of strcmp, it just comes from the std. C++ lib (which, admittedly, comes with our toolchain). If there is an error in the libs shipped with our toolchain, I'd like to find it-- when was the first time you noticed this issue?